index.bs

```diff
@@ -1327,14 +1264,231 @@ these steps:
+    the website wants the user to select. See [[#api-filpickeroptions-types]] for details.
+</div>
+
+<div algorithm>
+The <dfn method for=Window>showSaveFilePicker(|options|)</dfn> method, when invoked, must run
+these steps:
+
+1. Let |accepts options| be the result of [=process accept types|processing accept types=] given |options|.
+
+1. Let |environment| be <b>[=this=]</b>'s [=relevant settings object=].
+1. Let |global| be |environment|'s [=environment settings object/global object=].
+1. Verify that |environment| [=is allowed to show a file picker=].
+
+1. Let |p| be [=a new promise=].
+1. Run the following steps [=in parallel=]:
+
+  1. Optionally, wait until any prior execution of this algorithm has terminated.
+
+  1. Display a prompt to the user requesting that the user pick exactly one file.
+
+     The displayed prompt should let the user pick one of the |accepts options| to filter the list of displayed files.
+     Exactly how this is implemented, and what this prompt looks like is left up to the user agent.
+
+  1. Wait for the user to have made their selection.
+
+  1. If the user dismissed the prompt without making a selection,
+     [=/reject=] |p| with an {{AbortError}} and abort.
+
+  1. Let |entry| be a [=file entry=] representing the selected file.
+
+  1. If |entry| is deemed [=too sensitive or dangerous=] to be exposed to this website by the user agent:
+    1. Inform the user that the selected files or directories can't be exposed to this website.
+    1. At the discretion of the user agent,
+       either go back to the beginning of these [=in parallel=] steps,
+       or [=/reject=] |p| with an {{AbortError}} and abort.
+
+  1. Set |entry|'s [=query permission steps=] to the [=native file system query permission steps=].
+  1. Set |entry|'s [=request permission steps=] to the [=native file system request permission steps=].
+
+  1. Set |entry|'s [=binary data=] to an empty [=byte sequence=].
+
+  1. Set |result| to a new {{FileSystemFileHandle}} associated with |entry|.
+
+  1. Perform the <a spec=html>activation notification</a> steps in |global|'s [=Window/browsing context=].
+
+     Note: This lets a website immediately perform operations on the returned handles that
+     might require user activation, such as requesting more permissions.
+
+  1. [=/Resolve=] |p| with |result|.
+
+1. Return |p|.
+
+</div>
+
+## {{FilePickerOptions}}.{{FilePickerOptions/types}} ## {#api-filpickeroptions-types}
+
+<div class="note domintro">
+The {{showFilePicker(options)}} and {{showSaveFilePicker(options)}} methods accept a
+{{FilePickerOptions}} argument, which lets the website specify the types of files
+the file picker will let the user select.
+
+Each entry in {{FilePickerOptions/types}} specifies a single user selectable option
+for filtering the files displayed in the file picker.
+
+Each option consists of an <span class=allow-2119>optional</span> {{FilePickerAcceptType/description}}
```

Comment:
I think we're introducing "description" for the first time. Two concerns that can be addressed in follow-ups:

1) Description is the first user-visible (I assume?) bit of info in this section. Do we need a section on internalization concerns now?
2) Do we want non-normative advice on how the description would be used in the file picker? At least in Chrome, I think we try to avoid showing app-supplied strings in system UI (like file pickers).

Reply:
The old API had the same, but yeah, we definitely should have such a section. Filed #186 and #187.

Good point about displaying app-supplied strings in system UI. We've been doing that all along, and nobody has complained yet, but would probably be good to say something about in the spec. Filed #188 to track that.